agent/engine/engine_integ_test.go

```diff
@@ -215,51 +234,65 @@ func TestStartStopWithCredentials(t *testing.T) {
 	credentialsManager.SetTaskCredentials(taskCredentials)
 	testTask.SetCredentialsID(credentialsIDIntegTest)
 
-	taskEvents, contEvents := taskEngine.TaskEvents()
-
-	defer discardEvents(contEvents)()
+	stateChangeEvents := taskEngine.StateChangeEvents()
 
 	go taskEngine.AddTask(testTask)
 
-	verifyTaskIsStopped(taskEvents, testTask)
+	event := <-stateChangeEvents
+	assert.Equal(t, event.(api.ContainerStateChange).Status, api.ContainerRunning, "Expected container to be RUNNING")
+
+	event = <-stateChangeEvents
+	assert.Equal(t, event.(api.TaskStateChange).Status, api.TaskRunning, "Expected task to be RUNNING")
+
+	event = <-stateChangeEvents
+	assert.Equal(t, event.(api.ContainerStateChange).Status, api.ContainerStopped, "Expected container to be STOPPED")
+
+	event = <-stateChangeEvents
+	assert.Equal(t, event.(api.TaskStateChange).Status, api.TaskStopped, "Expected task to be STOPPED")
 
 	// When task is stopped, credentials should have been removed for the
 	// credentials id set in the task
 	_, ok := credentialsManager.GetTaskCredentials(credentialsIDIntegTest)
 	assert.False(t, ok, "Credentials not removed from credentials manager for stopped task")
 }
 
-func verifyTaskIsRunning(taskEvents <-chan api.TaskStateChange, testTasks ...*api.Task) error {
+func verifyTaskIsRunning(stateChangeEvents <-chan statechange.Event, testTasks ...*api.Task) error {
 	for {
 		select {
-		case taskEvent := <-taskEvents:
-			for i, task := range testTasks {
-				if taskEvent.TaskArn != task.Arn {
-					continue
-				}
-				if taskEvent.Status == api.TaskRunning {
-					if len(testTasks) == 1 {
-						return nil
+		case event := <-stateChangeEvents:
+			if event.GetEventType() == statechange.TaskEvent {
+				taskEvent := event.(api.TaskStateChange)
+				for i, task := range testTasks {
+					if taskEvent.TaskArn != task.Arn {
+						continue
+					}
+					if taskEvent.Status == api.TaskRunning {
+						if len(testTasks) == 1 {
+							return nil
+						}
+						testTasks = append(testTasks[:i], testTasks[i+1:]...)
+					} else if taskEvent.Status > api.TaskRunning {
+						return fmt.Errorf("Task went straight to %s without running, task: %s", taskEvent.Status.String(), task.Arn)
 					}
-					testTasks = append(testTasks[:i], testTasks[i+1:]...)
-				} else if taskEvent.Status > api.TaskRunning {
-					return fmt.Errorf("Task went straight to %s without running, task: %s", taskEvent.Status.String(), task.Arn)
 				}
 			}
 		}
 	}
 }
 
-func verifyTaskIsStopped(taskEvents <-chan api.TaskStateChange, testTasks ...*api.Task) {
+func verifyTaskIsStopped(stateChangeEvents <-chan statechange.Event, testTasks ...*api.Task) {
 	for {
 		select {
-		case taskEvent := <-taskEvents:
-			for i, task := range testTasks {
-				if taskEvent.TaskArn == task.Arn && taskEvent.Status >= api.TaskStopped {
-					if len(testTasks) == 1 {
-						return
+		case event := <-stateChangeEvents:
+			if event.GetEventType() == statechange.TaskEvent {
```

Comment:
I think the indentation is a bit high here.  How about reversing the conditional such that the normal case is unindented?

```go
if event.GetEventType() != statechange.TaskEvent {
	continue
}
```